service/controller/v5/resource/deployment/deployment.go

```diff
@@ -27,7 +27,7 @@ func (r Resource) newDeployment(ctx context.Context, customObject providerv1alph
 		"kubernetesAPISecurePort": key.APISecurePort(customObject),
 		"masterSubnetCidr":        sc.AzureNetwork.Master.String(),
 		"storageAccountName":      key.StorageAccountName(customObject),
-		"templatesBaseURI":        baseTemplateURI(r.templateVersion),
+		"templatesBaseURI":        key.ARMTemplateBaseURI(r.templateVersion, "deployment"),
```

Comment:
When we are here we can use this in the template instead:

```
 "vmssTemplateUrl": "[uri(deployment().properties.templateLink.uri, 'vmss.json')]"
```

See: https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-linked-templates#using-variables-to-link-templates

Reply:
Nice simplification